webpack/components/AnsibleHostDetail/components/JobsTab/NewRecurringJobHelper.js

```diff
@@ -0,0 +1,100 @@
+import * as Yup from 'yup';
+import { translate as __, sprintf } from 'foremanReact/common/I18n';
+
+import { showToast } from '../../../../toastHelper';
+
+export const frequencyOpts = [
+  { id: 'hourly', name: __('hourly') },
+  { id: 'daily', name: __('daily') },
+  { id: 'weekly', name: __('weekly') },
+  { id: 'monthly', name: __('monthly') },
+];
+
+export const rangeValidator = date => {
+  if (date < new Date()) {
+    return __('Must not be in the past');
+  }
+  return '';
+};
+
+export const createValidationSchema = () => {
+  const cantBeBlank = __("can't be blank");
+
+  return Yup.object().shape({
+    repeat: Yup.string().required(cantBeBlank),
+    startTime: Yup.string().required(cantBeBlank),
+    startDate: Yup.string().required(cantBeBlank),
+  });
+};
+
+export const toCron = (date, repeat) => {
+  switch (repeat) {
+    case 'hourly':
+      return `${date.getMinutes()} * * * *`;
+    case 'daily':
+      return `${date.getMinutes()} ${date.getHours()} * * *`;
+    case 'weekly':
+      return `${date.getMinutes()} ${date.getHours()} * * ${date.getDay()}`;
+    case 'monthly':
+      return `${date.getMinutes()} ${date.getHours()} ${date.getDate()} * *`;
+    default:
+      return `${date.getMinutes()} * * * *`;
+  }
+};
+
+export const toVars = (hostId, date, repeat) => ({
+  variables: {
+    jobInvocation: {
+      hostIds: [hostId],
+      feature: 'ansible_run_host',
+      targetingType: 'static_query',
+      scheduling: {
+        startAt: date,
+      },
+      recurrence: {
+        cronLine: toCron(date, repeat),
+        purpose: `ansible-host-${hostId}`,
```

Comment:
Could we have a function, that would return the purpose, I see it quite dangerous to have it in two places and rely on them being changed at the same time.

Reply:
I agree, updated.